hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/KeyGenUtils.java

```diff
@@ -74,7 +74,7 @@ public static String getPartitionPathFromGenericRecord(GenericRecord genericReco
   public static String[] extractRecordKeys(String recordKey) {
     String[] fieldKV = recordKey.split(",");
     return Arrays.stream(fieldKV).map(kv -> {
-      final String[] kvArray = kv.split(":");
+      final String[] kvArray = kv.split(":", 2);
```

Comment:
I assume this is needed when the value contains a colon (:).  Could you add a unit test for that?

Reply:
<img width="720" alt="image" src="https://user-images.githubusercontent.com/8789291/189589596-5a5dc805-4c01-4890-acf7-9a75e3eadb51.png">

Reply:
Yes, I was referring to the same case and asking for adding a unit test in the repo.  I added a unit for this PR.  Will land it once CI passes.